### DISCUSSION TITLE
The number of pulses wanted does not match the actual

### DISCUSSION THREAD

**Discussion by boxuan-zhao** [MrYsLab/telemetrix#48] [Q&A] [answered]
hello!
I want to use an Arduino Maga2560 to drive a step motor. when testing with ' stepper_absolute.py', I found that the angle of the stepper motor rotation was not expected. I tested it with a pulse counter and the number of pulses was 1/4 of expected. Is there any problem with my settings?

**Comment by MrYsLab** [comment, marked as answer]
@githubzbx You did not mention the motor and motor driver being used, nor did you supply your code, so it is difficult to determine the cause of your issue. Things to check are that when you call set_pin_mode_stepper, the interface type selected is correct. For example, when using a TB6600-like driver, the interface type is 1. Also, if you use a TB600-like driver, the microstep selection switches are correctly set on the driver.

There is an [excellent article](https://www.makerguides.com/tb6600-stepper-motor-driver-arduino-tutorial/) on interfacing stepper motors with an Arduino. There is a section about using [AccelStepper](https://github.com/waspinator/AccelStepper), the library that Telemetirx uses.

You might try running the example in the article or some of the examples in the Accelstepper library to take Telemetrix out of the equation and see if you are getting different results.

Please let me know if this answers your question or not.
